packages/@ember/-internals/container/lib/registry.ts

```diff
@@ -79,12 +79,12 @@ export default class Registry implements IRegistry {
   readonly _failSet: Set<string>;
   resolver: Resolver | (Resolve & NotResolver) | null;
   readonly fallback: IRegistry | null;
-  readonly registrations: { [key: string]: object };
-  _localLookupCache: { [key: string]: object };
-  readonly _normalizeCache: { [key: string]: string };
-  readonly _options: { [key: string]: TypeOptions };
-  readonly _resolveCache: { [key: string]: object };
-  readonly _typeOptions: { [key: string]: TypeOptions };
+  readonly registrations: { [key: string]: object | undefined };
+  _localLookupCache: { [key: string]: object | undefined };
+  readonly _normalizeCache: { [key: string]: string | undefined };
+  readonly _options: { [key: string]: TypeOptions | undefined };
+  readonly _resolveCache: { [key: string]: object | undefined };
+  readonly _typeOptions: { [key: string]: TypeOptions | undefined };
```

Comment:
1. Can we just use `Record<string, TypeOptions>` etc. here?
2. It looks like this as well as the `in` check below are both because we don't yet have `noUncheckedIndexAccess` enabled, so we're sort of pushing that safety in manually?

Reply:
We do have noUncheckedIndexAccess now. I’ll revisit this.